Sources/NIOHTTP2/HTTP2Error.swift

```diff
@@ -60,8 +60,16 @@ public enum NIOHTTP2Errors {
     }
 
     /// A stream state transition was attempted that was not valid.
-    public struct BadStreamStateTransition: NIOHTTP2Error {
-        public init() { }
```

Comment:
This empty public `init` cannot be removed.

With that in mind, let's go back to making `NIOHTTP2StreamState` public, and add `public` visibility to `BadStreamStateTransition.fromState`. Then, `fromState` can be optional and we can default it to `nil` for backwards compatibility reasons.

Reply:
ok, I made those changes.